File: backend/http/types/sourcecontrol/member_metrics.go
package sourcecontrol

// GetMemberMetricsRequest represents the request parameters for getting member metrics
type GetMemberMetricsRequest struct {
	StartDate string `form:"startDate" binding:"omitempty"` // YYYY-MM-DD format
	EndDate   string `form:"endDate" binding:"omitempty"`   // YYYY-MM-DD format
	Interval  string `form:"interval" binding:"omitempty"`  // daily, weekly, monthly
}

// SnapshotMetric represents a single metric in the snapshot
type SnapshotMetric struct {
	Label      string  `json:"label"`
	Value      float64 `json:"value"`
	PeersValue float64 `json:"peersValue"`
	Unit       string  `json:"unit"` // "count", "time" etc.
}

// SnapshotCategory represents a category of metrics in the snapshot
type SnapshotCategory struct {
	Category string           `json:"category"`
	Metrics  []SnapshotMetric `json:"metrics"`
}

// TimeSeriesDataPoint represents a single data point in a time series
type TimeSeriesDataPoint struct {
	Key   string  `json:"key"`
	Value float64 `json:"value"`
}

// TimeSeriesEntry represents a single time entry in a time series
type TimeSeriesEntry struct {
	Date string                `json:"date"`
	Data []TimeSeriesDataPoint `json:"data"`
}

// GraphMetric represents a single metric in the graph data
type GraphMetric struct {
	Label      string            `json:"label"`
	Type       string            `json:"type"`
	TimeSeries []TimeSeriesEntry `json:"timeSeries"`
}

// GraphCategory represents a category of metrics in the graph data
type GraphCategory struct {
	Category string        `json:"category"`
	Metrics  []GraphMetric `json:"metrics"`
}

// GetMemberMetricsResponse represents the response for getting member metrics
type GetMemberMetricsResponse struct {
	SnapshotMetrics []SnapshotCategory `json:"snapshotMetrics"`
	GraphMetrics    []GraphCategory    `json:"graphMetrics"`
}

File: frontend/src/types/memberMetrics.ts
// Request parameters for getting member metrics
export interface GetMemberMetricsParams {
  startDate?: string // YYYY-MM-DD format
  endDate?: string   // YYYY-MM-DD format
  interval?: string  // daily, weekly, monthly
}

// Snapshot metric for comparison
export interface SnapshotMetric {
  label: string
  description: string
  value: number
  peers_value: number
  unit: string // "count", "time", "loc", etc.
  icon_identifier: string
  icon_color: string
}

// Metric rule category
export interface MetricRuleCategory {
  name: string
  priority: number
}

// Category of snapshot metrics
export interface SnapshotCategory {
  category: MetricRuleCategory
  metrics: SnapshotMetric[]
}

// Time series data point
export interface TimeSeriesDataPoint {
  key: string
  value: number
}

// Time series entry
export interface TimeSeriesEntry {
  date: string
  data: TimeSeriesDataPoint[]
}

// Graph metric for visualization
export interface GraphMetric {
  label: string
  type: string
  category?: string
  unit?: string
  time_series: TimeSeriesEntry[]
}

// Category of graph metrics
export interface GraphCategory {
  category: MetricRuleCategory
  metrics: GraphMetric[]
}

// Response for member metrics
export interface GetMemberMetricsResponse {
  snapshot_metrics: SnapshotCategory[]
  graph_metrics: GraphCategory[]
} 